Unity-Technologies/ScriptableRenderPipeline
# Psvr fixes

### Purpose of this PR

Fix several bugs that show up when testing HDRP in VR on PlayStation VR and PS4.
* Tessellation shaders in VR will not compile on PS4.
* The CameraNode direction property in the shadergraph relied on implicit truncation from a float4x4 to a float3x3 which cases an error on PS4.

---
### Testing status

**Manual Tests**: What did you do?
- [X] Ran the VR test project on PSVR to verify the issues were fixed.
- Other: 

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline
Ran the following: _ABV for SRP repository - fast-trunk

5 Jobs failed on this branch:
* [HDRP on Win_DX11_playmode on version fast-trunk](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/psvr-fixes/.yamato%252Fupm-ci-abv.yml%2523all_project_ci_fast-trunk/822350/job/pipeline)
* [HDRP on Win_DX11_playmode XR on version fast-trunk](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/psvr-fixes/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Win_DX11_playmode_fast-trunk/822337/job/pipeline)

These both fail on `HDRP_GraphicTestRunner.2008_Light_Volumetrics`. This test is also failing on the master branch at the moment see https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/master/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Win_DX11_playmode_fast-trunk/831034/job

XR version has lots of failure on both this version and [master](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/master/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Win_DX11_playmode%2520XR_fast-trunk/868512/job)

* [VFX_LWRP on Win_DX11_Standalone on version fast-trunk](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/psvr-fixes/.yamato%252Fupm-ci-vfx_lwrp.yml%2523VFX_LWRP_Win_DX11_Standalone_fast-trunk/822347/job/pipeline)
* [VFX_LWRP on Win_DX11_playmode on version fast-trunk](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/psvr-fixes/.yamato%252Fupm-ci-vfx_lwrp.yml%2523VFX_LWRP_Win_DX11_playmode_fast-trunk/822348/job/pipeline)

These are also failing on master.

* [VFXMain on Win_DX11_playmode on version fast-trunk](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/psvr-fixes/.yamato%252Fupm-ci-vfxmain.yml%2523VFXMain_Win_DX11_playmode_fast-trunk/822344/job/pipeline) - [also fails on master](https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/master/.yamato%252Fupm-ci-vfxmain.yml%2523VFXMain_Win_DX11_playmode_fast-trunk/868372/job)

---
### Comments to reviewers
Notes for the reviewers you have assigned.
